packages/shared-ux/page/solution_nav/src/solution_nav.tsx

```diff
@@ -174,6 +176,32 @@ export const SolutionNav: FC<SolutionNavProps> = ({
     );
   }, [children, headingID, isCustomSideNav, isHidden, items, rest]);
 
+  const navWidth = useMemo(() => {
+    if (isLargerBreakpoint) {
+      return isOpenOnDesktop ? `${FLYOUT_SIZE}px` : euiThemeVars.euiSizeXXL;
+    }
+    if (isMediumBreakpoint) {
+      return isSideNavOpenOnMobile || !canBeCollapsed
+        ? `${FLYOUT_SIZE}px`
+        : euiThemeVars.euiSizeXXL;
+    }
+    return '0';
+  }, [
+    isOpenOnDesktop,
+    isSideNavOpenOnMobile,
+    canBeCollapsed,
+    isMediumBreakpoint,
+    isLargerBreakpoint,
+  ]);
+  const { setGlobalCSSVariables } = useEuiThemeCSSVariables();
+  // Setting a global CSS variable with the nav width
+  // so that other pages have it available when needed.
+  useEffect(() => {
+    setGlobalCSSVariables({
+      '--kbnSolutionNavOffset': navWidth,
+    });
```

Comment:
I guess this should be cleaned in the destroy callback?

Reply:
I had that initially, but it didn't seem to change anything so I removed it as superfluous.

Reply:
it could be a problem when this variable is used with a fallback for the case when solution nav is not rendered, e.g.: 

```
var(--kbnSolutionNavOffset, 0);
```